src/Ocelot/Multiplexer/MultiplexingMiddleware.cs

```diff
@@ -255,5 +255,31 @@ protected virtual Task MapAsync(HttpContext httpContext, Route route, List<HttpC
 
         var aggregator = _factory.Get(route);
         return aggregator.Aggregate(route, httpContext, contexts);
+    }
+
+    protected virtual async Task<Stream> CloneRequestBodyAsync(HttpRequest request, CancellationToken aborted)
+    {
+        request.EnableBuffering();
+        if (request.Body.Position == 0)
+        {
+            var targetBuffer = new MemoryStream();
+            if (request.ContentLength is not null)
+            {
+                await request.Body.CopyToAsync(targetBuffer, (int)request.ContentLength, aborted);
+                targetBuffer.Position = 0;
+                request.Body.Position = 0;
+            }
+            else
+            {
+                Logger.LogWarning("Aggregation does not support body copy without Content-Length header!");
+            }
+
+            return targetBuffer;
+        }
+        else
+        {
+            Logger.LogWarning("Ocelot does not support body copy without stream in initial position 0");

```

Comment:
By the way, should this possibly happen? I'm wondering if we should throw here...

Reply:
I'd keep this logged warning just for debugging and proving our solution with the fix is functioning.
So, we have improved version of multiplexer but it is draft from the point of view of Production environment.
Let's keep this `else` branch!

Reply:
@ggnaegi I don't think it should happen in a normal scenario. The only way I can think of is Ocelot reading partially the body elsewhere by mistake.

However any type of custom middleware could potentially be reading a partial body. I don't see any use case right now but I think it's preferable to log instead of throwing an exception.

Reply:
Not an issue! Resolving...